src/atom-environment.coffee

```diff
@@ -678,6 +684,7 @@ class AtomEnvironment extends Model
         @deserialize(state) if state?
         @deserializeTimings.atom = Date.now() - startTime
 
+        @document.body.appendChild(@views.getView(@titleBar)) if @titleBar
```

Comment:
> I thought the combined titlebar might primarily be hosting the tab-bar, which might be pragmatic for a start, toolbars do sound funky though.

Only visually maybe, but tab-bars have to live inside panes. Because they can be split into different directions and each pane as its own tab-bar.

Anyways, I think for now it would be safer to add the title-bar into the .header. Because the changes to `workspace-view.less` might invite other packages to add their bars or other UI stuff outside of `atom-workspace`. With the `addXXXPanel()` all panels can be kept a bit more under control.